### DISCUSSION TITLE
World Generation, Is there a way with the GenerationUnit's UnitModifier to only replace air ?

### DISCUSSION THREAD

**Discussion by BelgianDev** [Minestom/Minestom#2478] [Help]
I'm messing with world gen and I'm wondering if there this a way to replace only air or another specific block.
Or if the fill method only replaces air ?

Or else I could use the `#setAllRelative` method and check manually for the block type but I'm worried it could do the opposite that I'm trying to achieve and instead slow down generation

**Comment by KrystilizeNevaDies** [comment]
There is no way to know if the position you are placing blocks at have an air block or any other type of block.
If you want to do some set operation like this, I suggest doing the set operation before the rest of your worldgen, or build a wrapper around minestom's api.
